<!-- -*- mode: XML -*- -->

<launch>
    <node name="dynamixel_manager" pkg="dynamixel_hardware_interface" type="controller_manager" output="screen" >
        <rosparam file="$(find wubble2_robot)/config/dynamixel_ports.yaml" command="load"/>
    </node>
    
    <!-- Start arm single joint controllers -->
    <rosparam file="$(find wubble2_robot)/config/wubble_arm.yaml" command="load"/>
    <node name="upper_arm_spawner" pkg="dynamixel_hardware_interface" type="controller_spawner.py"
          args="--manager=wubble_motor_manager
                --port=wubble2UpperArm
                shoulder_pitch_controller
                shoulder_pan_controller
                upperarm_roll_controller
                elbow_flex_controller"
          output="screen"/>
          
    <node name="lower_arm_spawner" pkg="dynamixel_hardware_interface" type="controller_spawner.py"
          args="--manager=wubble_motor_manager
                --port=wubble2LowerArm
                forearm_roll_controller
                wrist_pitch_controller
                wrist_roll_controller"
          output="screen"/>
          
    <!-- Start arm trajectory action controller -->
    <rosparam file="$(find wubble2_robot)/config/wubble_arm_trajectory_controller.yaml" command="load"/>
    <node name="arm_trajectory_controller_spawner" pkg="dynamixel_hardware_interface" type="controller_spawner.py"
          args="--manager=wubble_motor_manager
                --port=multi_joint_dummy_port
                l_arm_controller"
          output="screen"/>
          
     <!-- Start head and neck single joint controllers -->
    <rosparam file="$(find wubble2_robot)/config/wubble_head.yaml" command="load"/>
    <node name="head_controller_spawner" pkg="dynamixel_hardware_interface" type="controller_spawner.py"
          args="--manager=wubble_motor_manager
                --port=wubble2Head
                head_pan_controller
                head_tilt_controller"
          output="screen"/>
          
    <node name="neck_controller_spawner" pkg="dynamixel_hardware_interface" type="controller_spawner.py"
          args="--manager=wubble_motor_manager
                --port=wubble2Neck
                neck_tilt_controller"
          output="screen"/>
</launch>

